database/sql/src/main/scala/cromwell/database/slick/tables/CallCachingAggregationEntryComponent.scala

```diff
@@ -50,6 +50,20 @@ trait CallCachingAggregationEntryComponent {
     } yield ()).exists
   )
 
+  val existsCallCachingEntriesForBaseAggregationHashWithExecutionBucket = Compiled(
+    (baseAggregation: Rep[String], executionBucket: Rep[String], executionBucketLength: Rep[Int]) => (for {
+      callCachingEntry <- callCachingEntries
+      if callCachingEntry.allowResultReuse
+      callCachingAggregationEntry <- callCachingAggregationEntries
+      if callCachingEntry.callCachingEntryId === callCachingAggregationEntry.callCachingEntryId
+      if callCachingAggregationEntry.baseAggregation === baseAggregation
+      detritus <- callCachingDetritusEntries
```

Comment:
Is this checking that all detrituses have the bucket prefix ?

Reply:
This is meant to check that there exists a detritus file that belongs to this call caching entry whose GCS bucket prefix matches the desired value.

Reply:
It'll be interesting to see if this extra join has any effect on the DB performance of reading cache hits. Even if it does the overall gain will probably be enormous anyway but would be good to know with the incoming perf testing